Snowflake.Data/Core/SnowflakeTomlConnectionBuilder.cs

```diff
@@ -0,0 +1,128 @@
+// <copyright file="SnowflakeTomlConnectionBuilder.cs" company="Snowflake Inc">
+//         Copyright (c) 2024 Snowflake Inc. All rights reserved.
+//  </copyright>
+
+namespace Snowflake.Data.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using Client;
+    using Log;
+    using Tomlyn;
+    using Tomlyn.Model;
+    using Tools;
+
+    public class SnowflakeTomlConnectionBuilder
+    {
+        private const string DefaultConnectionName = "default";
+        private const string DefaultSnowflakeFolder = ".snowflake";
+        private const string DefaultTokenPath = "/snowflake/session/token";
+
+        private readonly SFLogger _logger = SFLoggerFactory.GetLogger<SnowflakeDbConnection>();
+
+        private readonly Dictionary<string, string> _tomlToNetPropertiesMapper = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase)
+        {
+            { "DATABASE", "DB" }
+        };
+
+        private readonly FileOperations _fileOperations;
+        private readonly EnvironmentOperations _environmentOperations;
+
+        public SnowflakeTomlConnectionBuilder() : this(FileOperations.Instance, EnvironmentOperations.Instance)
+        {
+        }
+
+        internal SnowflakeTomlConnectionBuilder(FileOperations fileOperations, EnvironmentOperations environmentOperations)
+        {
+            _fileOperations = fileOperations;
+            _environmentOperations = environmentOperations;
+        }
+
+        public string GetConnectionStringFromToml(string connectionName = null)
+        {
+            var connectionString = string.Empty;
+            var tomlPath = ResolveConnectionTomlFile();
+            var connectionToml = GetTomlTableFromConfig(tomlPath, connectionName);
+            if (connectionToml != null)
+            {
+                connectionString = GetConnectionStringFromTomlTable(connectionToml);
+            }
+            return connectionString;
+        }
+
+        private string GetConnectionStringFromTomlTable(TomlTable connectionToml)
+        {
+            var connectionStringBuilder = new StringBuilder();
+            var tokenFilePathValue = string.Empty;
+            var isOauth = connectionToml.TryGetValue("authenticator", out var authenticator) && authenticator.ToString().Equals("oauth");
+            foreach (var property in connectionToml.Keys)
+            {
+                if (isOauth && property.Equals("token_file_path", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    tokenFilePathValue = (string)connectionToml[property];
+                    continue;
+                }
+                var mappedProperty = _tomlToNetPropertiesMapper.TryGetValue(property, out var mapped) ? mapped : property;
+                connectionStringBuilder.Append($"{mappedProperty}={(string)connectionToml[property]};");
+            }
```

Comment:
Which is the current behavior of our driver to unknown parameters for the connection string, the implementation for the TomlConnectionBuilder is to translate the client configuration to an equivalent connection string. Maybe we should handle that scenario when processing the connection string instead of here. WDYT @sfc-gh-knozderko ?

Reply:
This logic is already handled by our SnowflakeDbConnection if a parameter unknown is send is ignored and logged.